src/course-outline/CourseOutline.jsx

```diff
@@ -117,6 +121,23 @@ const CourseOutline = ({ courseId }) => {
     errors,
   } = useCourseOutline({ courseId });
 
+  // Use `setToastMessage` to show the toast.
+  const [toastMessage, setToastMessage] = useState(/** @type{null|string} */ (null));
+
+  useEffect(() => {
+    if (location.hash === '#export-tags') {
+      setToastMessage(intl.formatMessage(messages.exportTagsCreatingToastMessage));
+      getTagsExportFile(courseId, courseName).then(() => {
+        setToastMessage(intl.formatMessage(messages.exportTagsSuccessToastMessage));
+      }).catch(() => {
+        setToastMessage(intl.formatMessage(messages.exportTagsErrorToastMessage));
+      });
+
+      // Delete `#export-tags` from location
+      window.location.href = '#';
+    }
+  }, [location]);
```

Comment:
I'm not sure why this URL change is being used as an indicator of this state. 

I think whatever triggers the tag exporting should set the toastMessage or some flag that an export is in progress.

Reply:
The `Export Tags` button is on the [StudioHeader](https://github.com/openedx/frontend-app-course-authoring/blob/bc63bfc7f585e3751e356eb213390ee916a742f6/src/header/Header.jsx#L47) [(frontend-component-header)](https://github.com/openedx/frontend-component-header). This component only receives as parameters the [text of the button and the link](https://github.com/openedx/frontend-app-course-authoring/pull/995/files#diff-e660e1566707a54f0764f507112244f0c929f18e3198814a49424507fac6198bL72) to where that button should redirect. 

There is no way to send an `onClick` function or know when the user clicked the button or add some flag to know if the export is in progress, this is necessary to display the toast. The component [logic is closed ](https://github.com/openedx/frontend-component-header/blob/master/src/studio-header/NavDropdownMenu.jsx#L22) in the link that is passed as a parameter.

My workaroun is to use an internal MFE route instead the edx-platform API. The detail is that there should be no redirection. To achieve this I have used the url fragment.  The useEffect of this code is to verify the fragment and execute the export code we want with the respective toasts.

Reply:
Sounds good! 

I think another mechanism could be to use postMessage, but I guess if all you can do is inject a link this is a good solutions.